### PR TITLE
chore(config): remove EnableLocalhostInboundClusters option

### PR DESCRIPTION
**Pull request by lobkovilya**
Cherry-pick of the https://github.com/kumahq/kuma/pull/6214 that was reverted.

Fix https://github.com/kumahq/kuma/issues/5335 https://github.com/kumahq/kuma/issues/4772

### Checklist prior to review

- [X] [Link to relevant issue][1] as well as docs and UI issues --
- [X] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [X] Tests (Unit test, E2E tests, manual test on universal and k8s) --
- [X] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [X] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? --
- [X] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

**Review by lukidzi — COMMENTED** [review]
We can add `upstream_bind_config` to point `sourceAddress: 127.0.0.6` in https://github.com/kumahq/kuma/blob/master/pkg/hds/tracker/healthcheck_generator.go#L133 so we can enable health check test on universal.

File: pkg/hds/tracker/healthcheck_generator.go [hunk 1]
```diff
@@ -137,7 +137,7 @@ func (g *SnapshotGenerator) envoyHealthCheck(port uint32) *envoy_service_health.
 				Address: &envoy_core.Address{
 					Address: &envoy_core.Address_SocketAddress{
 						SocketAddress: &envoy_core.SocketAddress{
-							Address: "127.0.0.1",
```
Comment (lukidzi) [inline pkg/hds/tracker/healthcheck_generator.go:140]: I mean to use https://www.envoyproxy.io/docs/envoy/v1.25.3/api-v3/service/health/v3/hds.proto.html#service-health-v3-clusterhealthcheck `upstream_bind_config `, we still want to do request to `127.0.0.1` but we want to override source address os iptables can handle it. https://github.com/kumahq/kuma/blob/master/pkg/xds/generator/inbound_proxy_generator.go#L43 like this so we don't do it for no TP deployment

Comment (lobkovilya) [reply pkg/hds/tracker/healthcheck_generator.go]: Right, fixed

**Review by lukidzi — APPROVED** [review]

**Pull request merged**